src/Components/Controller.js

```diff
@@ -53,29 +53,36 @@ class Controller extends React.Component {
       ...this.props,
       _bikeTrafficHeight: this._bikeTrafficHeight,
       _carTrafficHeight: this._carTrafficHeight
-    }
+    };
 
-    const visibleLayers = layers.setLayers( layerSettings );
+    const visibleLayers = layers.setLayers(layerSettings);
 
-    this.props.onLayerChange( visibleLayers );
+    this.props.onLayerChange(visibleLayers);
   };
 
-  _toggleLayer = ({ checked, key, animate, heightKey }) => {
+  _toggleLayer = ({ checked, key, animate, heightKey, ptMaster }) => {
     let newState = { ...this.props };
-    newState[ key ] = checked;
+    newState[key] = checked;
 
-    if( animate && heightKey ) {
-      layers.animateLayer(
-        v => ( this[ heightKey ] = v ),
-        checked,
-        () => this._updateLayers()
+    if (animate && heightKey) {
+      layers.animateLayer(v => (this[heightKey] = v), checked, () =>
+        this._updateLayers()
       );
     }
 
-    this.props.onChange( newState );
+    if (ptMaster) {
+      // Master checkbox have been pressed. Update all sub checkboxes.
```

Comment:
This is too specific. Let's make it generalized so we can have groups on any data types and it works consistently between them.

Also, we should talk about behavior:
-When the master is turned off, it should turn off all children
-When the master is turned on, it should turn on all children
-If the master is off, when one of the children is turned on, it should cause the master to turn on without turning on all of the children

That behavior should be defined in controlOption.js
We can talk about how that would work